mro/custom_types.py

```diff
@@ -0,0 +1,214 @@
+import mro
+import datetime
+import csv
+import psycopg2.extensions, psycopg2.extras
+from operator import attrgetter
+
+
+def _get_custom_type_names(connection):
+    cursor = connection.cursor()
+    # Get custom types
+    cursor.execute("""SELECT t.typname as type
+                                FROM pg_type t
+                                LEFT JOIN pg_catalog.pg_namespace n
+                                       ON n.oid = t.typnamespace
+                                WHERE 
+                                (
+                                      t.typrelid = 0 
+                                      OR 
+                                      (
+                                           SELECT c.relkind = 'c' 
+                                           FROM pg_catalog.pg_class c
+                                           WHERE c.oid = t.typrelid
+                                      )
+                                ) 
+                                AND NOT EXISTS
+                                (
+                                      SELECT 1
+                                      FROM pg_catalog.pg_type el 
+                                      WHERE el.oid = t.typelem
+                                      AND el.typarray = t.oid
+                                )
+                                AND n.nspname NOT IN 
+                                (
+                                      'pg_catalog',
+                                      'information_schema'
+                                )
+                        """)
+    connection.commit()
+    types = [t for t in cursor]
+    return types
+
+
+def customColumnToDataType(column, code_start, code_end):
+    python_type = f'mro.custom_types.{column[27]}'
+    return '{0}{1}, {2}'.format(code_start, python_type, code_end)
+
+
+def strtobool(value: str):
+    return {'true': True,
+            'false': False,
+            't': True,
+            'f': False
+            }.get(value.lower(), None)
+
+
+def strtodate(value: str):
+    return datetime.datetime.strptime(value, '%Y-%m-%d').date()
+
+
+def strtotime(value: str):
+    return datetime.datetime.strptime(value, '%H:%M:%S.%f').time()
+
+
+def strtotimestamp(value: str):
+    return datetime.datetime.strptime(value, '%Y-%m-%d %H:%M:%S.%f')
+
+
+postgres_type_to_python_map = {
+    'character varying': str,
+    'text': str,
+    'integer': int,
+    'boolean': strtobool,
+    'bool': strtobool,
+    'float': float,
+    'double precision': float,
+    'real': float,
+    'date': strtodate,
+    'timestamp without time zone': strtotimestamp,
+    'time without time zone': strtotime,
+    'json': str,
+    'jsonb': str,
+    'uuid': str,
+    'bytea': str,
+}
+
+
+class CustomFieldType:
+    def __init__(self, attribute):
```

Comment:
This is a wrapper around the internal fields of a custom type, this allows you to raise the error if you try to set it, in the future we can change the set to run the sql required to update the field in the db